pkg/provider/provider.go

```diff
@@ -167,6 +167,9 @@ func (k *kubeProvider) DiffConfig(ctx context.Context, req *pulumirpc.DiffReques
 	if olds["cluster"] != news["cluster"] {
 		diffs = append(diffs, "cluster")
 	}
+	if olds["namespace"] != news["namespace"] {
+		diffs = append(diffs, "namespace")
+	}
```

Comment:
Special case is that `"" == "default"`, right?

Reply:
Verified that this works as intended. It will show a change on the provider, but does not cause any changes to dependent resources.